server/dive_server/viame.py

```diff
@@ -378,7 +379,50 @@ def postprocess(self, folder, skipJobs):
 
     @access.user
     @autoDescribeRoute(
-        Description("").jsonParam("data", "", requireObject=True, paramType="body")
+        Description("")
+        .modelParam(
+            "folderId",
+            description="folder id of a clip",
+            model=Folder,
+            paramType="query",
+            required=True,
+            level=AccessType.WRITE,
+        )
+        .jsonParam(
+            "attributes",
+            "upsert and delete attributes",
+            paramType="body",
+            requireObject=True,
+        )
+    )
+    def save_attributes(self, folder, attributes):
+        upsert = attributes.get('upsert', [])
+        delete = attributes.get('delete', [])
+        attributes_dict = {}
+        if 'attributes' in folder['meta']:
+            attributes_dict = folder['meta']['attributes']
+        for attribute_id in delete:
+            attributes_dict.pop(str(attribute_id), None)
+        for attribute in upsert:
+            attributes_dict[str(attribute['_id'])] = attribute
```

Comment:
I don't know if we even need an `id` anymore.  The tracks attributes and detection attributes are limited by their name (because of the CSV file).  So really the identifier is a combination of the `belongs` and the name.